File: app/ios/NeuroPulse/Analytics/AnalyticsBackend.swift
import Foundation

/// Abstraction layer between `ResearchAnalyticsGate` and any concrete analytics SDK.
///
/// All PostHog API calls live exclusively in `PostHogAnalyticsBackend`.
/// Tests inject a `SpyAnalyticsBackend` (test target only) via
/// `ResearchAnalyticsGate._backend` — no test file ever imports PostHog directly (ISC-157).
@MainActor
protocol AnalyticsBackend: AnyObject {
    /// Initialise (or re-enable) the analytics SDK after the user has consented.
    func configure()

    /// Forward a pre-validated event to the SDK. Called only when the gate is open
    /// and all prohibited-key checks have already passed.
    func track(event: String, properties: [String: String])

    /// Opt the SDK out and reset the anonymous distinct ID on consent withdrawal.
    func reset()
}

File: app/android/core/src/main/kotlin/com/neurone/core/analytics/AnalyticsBackend.kt
package com.neurone.core.analytics

// Analytics backend abstraction — port of iOS AnalyticsBackend protocol.
// The concrete production backend (PostHog Android SDK) lives in :app;
// call sites use ResearchAnalyticsGate exclusively and never the SDK directly.

interface AnalyticsBackend {
    fun configure()
    fun reset()
    fun track(event: String, properties: Map<String, String>)
}
